app/components/app_vaccination_record_details_component.rb

```diff
@@ -9,6 +9,28 @@ def initialize(vaccination_record)
     @batch = vaccination_record.batch
   end
 
+  def dose_number
+    return nil if @vaccine.nil? || @vaccine.seasonal?
+
+    numbers_to_words = {
+      1 => "First",
+      2 => "Second",
+      3 => "Third",
+      4 => "Fourth",
+      5 => "Fifth",
+      6 => "Sixth",
+      7 => "Seventh",
+      8 => "Eighth",
+      9 => "Ninth"
+    }.freeze
+
+    if @vaccination_record.dose_sequence <= 9
+      numbers_to_words[@vaccination_record.dose_sequence]
+    else
+      @vaccination_record.dose_sequence
```

Comment:
Even if you prefer the words above, `.ordinalize` could work here.

```suggestion
      @vaccination_record.dose_sequence.ordinalize
```